src/graphql-project/internal/infrastructure/repository/employee.go

```diff
@@ -0,0 +1,92 @@
+package repository
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sheelendar196/go-projects/graphql-project/internal/core/domain"
+	"gorm.io/gorm"
+)
+
+type Employee struct {
+	list []*domain.Employee
+}
+
+func NewEmployeeRepo(gormDB *gorm.DB) *Employee {
+	return &Employee{}
+}
+
+func (e *Employee) SaveEmployee(ctx context.Context, employee *domain.Employee) error {
+	e.list = append(e.list, employee)
+	return nil
+}
+
+func (e *Employee) GetEmployeeByID(ctx context.Context, empID string) (*domain.Employee, error) {
+	for _, emp := range e.list {
+		if emp.EmpID == empID {
+			return emp, nil
+		}
+	}
+
+	return nil, errors.New("employee not found in db")
+}
+
+func (e *Employee) GetEmployeeList(ctx context.Context) ([]*domain.Employee, error) {
+	return e.list, nil
+}
+
+func (e *Employee) UpdateEmployeeDetails(ctx context.Context, input domain.Employee) (*domain.Employee, error) {
```

Comment:
These logic checks should happen in worker. Keep only the calls to the db in this layer

Reply:
UpdateEmployeeDetails update employee into table on once actual table will be there this logic removed. it is just search into table(list)